### ISSUE TITLE
Unproject problem

### ISSUE THREAD

**Issue by jacekolszak** [faiface/pixel#159]
I am struggling with unprojecting mouse position using camera transformation matrix. I am getting strange results when rotating the matrix. Here is a test which proves that either something is wrong or I don't understand math. Following test fails with version 0.8.0.

```go
package test

import (
	"github.com/faiface/pixel"
	"github.com/stretchr/testify/assert"
	"math"
	"testing"
)

func TestMatricUnproject(t *testing.T) {
	matrix := pixel.IM.Rotated(pixel.ZV, math.Pi/2)
	unprojected := matrix.Unproject(pixel.V(0, 1))
	assert.InDelta(t, unprojected.X, 1, 0.1)
	assert.InDelta(t, unprojected.Y, 0, 0.1)
}
```

**Comment by faiface** [comment]
Yep, seems like you're right! I've run this test on my computer:

```go
package main

import (
	"fmt"
	"github.com/faiface/pixel"
	"math"
)

func main() {
	matrix := pixel.IM.Rotated(pixel.ZV, math.Pi/2)
	fmt.Println(matrix.Unproject(pixel.V(1, 0)))
	fmt.Println(matrix.Project(pixel.V(1, 0)))
}
```

And both lines print the same value. That's a clear bug.

If you know how to fix it, or are willing to investigate, I'd be happy to accept a fix :). It should be easy, just put the correct formula for the inverse matrix in `Unproject`.

**Comment by jacekolszak** [comment]
Hi @faiface, I would love to help. There is one problem though - I am not a linear algebra expert. I can fix the problem based on some other implementation available in the Internet (such as mathgl module). But I am not sure if the solution will be as efficient as the current one. Here is a quick and dirty PoC which could potentially work:

```go
func (m Matrix) Unproject(u Vec) Vec {
	det := m[0]*m[3] - m[2]*m[1]
	return Vec{
		m[3]/det*u.X - m[2]/det*u.Y + m[2]*m[5] - m[3]*m[4],
		-m[1]/det*u.X + m[0]/det*u.Y + m[1]*m[4] - m[0]*m[5],
	}
}
```
I ran benchmarks for this piece of code and found out that it is order of magnitude slower than existing implementation.

Nevertheless I can try to fix it. First of all `Unproject()` method is not unit-tested at all. So I can start with writing some unit tests first. What do you say about adding `testify` to the list of project dependencies? I just can't imagine writing unit tests without proper assertion framework. Do you know a better alternative or would you like to avoid adding any dependencies?

**Comment by faiface** [comment]
Hi, no worries, we'll figure it out :)

For example, you use 4 divisions in the PoC. But they can be easily reduced to just 2 divisions. Just replace `m[3]/det*u.X - m[2]/det*u.Y` with `(m[3]*u.X - m[2]*u.Y)/det`.

Regarding tests, feel free to include `testify` if that's what you like working with, no problem.

**Comment by faiface** [comment]
Fixed by #160
